sdk/src/test/java/com/mapbox/maps/ResourceAttributeParserTest.kt

```diff
@@ -29,28 +30,34 @@ class ResourceAttributeParserTest {
     every { typedArray.hasValue(any()) } returns true
   }
 
+  @After
+  fun cleanUp() {
+    ResourceOptionsManager.destroyDefault()
+  }
+
   @Test
   fun default() {
     val resourceOptions =
-      ResourcesAttributeParser.parseResourcesOptions(context, typedArray, CredentialsManager.default)
+      ResourcesAttributeParser.parseResourcesOptions(context, typedArray)
     assertEquals("pk.foobar", resourceOptions.accessToken)
     assertEquals(null, resourceOptions.baseURL)
     assertEquals("/foobar/.mapbox/maps/ambient_cache.db", resourceOptions.cachePath)
     assertEquals(99L, resourceOptions.cacheSize)
   }
 
-  @Test(expected = MapboxConfigurationException::class)
-  fun noAccessTokenResource() {
+  @Test
+  fun noAccessTokenResourceHasTokenFromAttr() {
```

Comment:
shall we add a test that `noAccessTokenResourcesNoTokenFromAttr` and `hasAccessTokenResourcesHasTokenFromAttr` and `hasAccessTokenResourcesNoTokenFromAttr` ? seems a bit complicated, but I guess worth to add it since the logic(decision tree) here is complex.